integration/apidefinition_test.go

```diff
@@ -0,0 +1,150 @@
+	"github.com/matryer/is"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+const (
+	apiDefWithJSONValidationName = "apidef-json-validation"
+	apiDefListenPath             = "/validation"
+	defaultVersion               = "Default"
+	errorResponseCode            = 422
+	defaultTimeout               = 1 * time.Minute
+)
+
+func TestApiDefinitionCreate(t *testing.T) {
+	eps := &model.ExtendedPathsSet{
+		ValidateJSON: []model.ValidatePathMeta{{
+			ErrorResponseCode: errorResponseCode,
+			Path:              "/get",
+			Method:            http.MethodGet,
+			Schema: &model.JSONValidationSchema{Unstructured: unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"properties": map[string]interface{}{
+						"key": map[string]interface{}{
+							"type":      "string",
+							"minLength": 2,
+						},
+					},
+				},
+			}},
+		}},
+	}
+
+	adCreate := features.New("ApiDefinition").
+		Setup(func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+			testNS := ctx.Value(ctxNSKey).(string) //nolint:errcheck
+			is := is.New(t)
+
+			// Create ApiDefinition with JSON Schema Validation support.
+			_, err := createTestAPIDef(ctx, testNS, func(apiDef *v1alpha1.ApiDefinition) {
+				apiDef.Name = apiDefWithJSONValidationName
+				apiDef.Spec.Proxy = model.Proxy{
+					ListenPath:      apiDefListenPath,
+					TargetURL:       "http://httpbin.org",
+					StripListenPath: true,
+				}
+				apiDef.Spec.VersionData.DefaultVersion = defaultVersion
+				apiDef.Spec.VersionData.NotVersioned = true
+				apiDef.Spec.VersionData.Versions = map[string]model.VersionInfo{
+					defaultVersion: {Name: defaultVersion, UseExtendedPaths: true, ExtendedPaths: eps},
+				}
+			}, envConf)
+			is.NoErr(err) // failed to create apiDefinition
+
+			return ctx
+		}).
+		Assess("ApiDefinition must have ValidateJSON field",
+			func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+				is := is.New(t)
+				client := cfg.Client()
+
+				testNS := ctx.Value(ctxNSKey).(string) //nolint:errcheck
+				desiredApiDef := v1alpha1.ApiDefinition{
+					ObjectMeta: metav1.ObjectMeta{Name: apiDefWithJSONValidationName, Namespace: testNS},
+				}
+
+				err := wait.For(conditions.New(client.Resources()).ResourceMatch(&desiredApiDef, func(object k8s.Object) bool {
+					apiDef := object.(*v1alpha1.ApiDefinition) //nolint:errcheck
+					// 'validate_json' field must exist in the ApiDefinition object.
+					return len(apiDef.Spec.VersionData.Versions[defaultVersion].ExtendedPaths.ValidateJSON) == 1
+				}), wait.WithTimeout(defaultTimeout))
+				is.NoErr(err)
+
+				return ctx
+			}).
+		Assess("ApiDefinition must exists in the Gateway",
```

Comment:
Since you are sending request to API in the next assessment can this be removed?